# Analysis of Chicago Communities with SQL



# Introduction

This project uses SQL queries to analyzes three datasets about the communities in the city of Chicago. The datasets are in csv format. Thus, first, database tables were created on IBM Db2 on Cloud and then the datasets were loaded to those database tables. Next, the connection to the database was made to begin the SQL queries. The results of SQL queries help answer questions like what types of crimes were recorded at different levels of schools, which communities have more crimes, which community is the most prone to crimes, which communities have lower income per capita than a set level, and which community has the highest percentage of households below poverty line etc. Overall, this analysis offers a glance of the socioeconomic status and the crime situations of the city of Chicago as well as the crimes related issues at its schools.  


Author: Avery Jan

Date: 4-3-2022


# Datasets

This project uses the subsets of three datasets available on the city of Chicago's Data Portal. 

1. Socioeconomic Indicators in Chicago Subset (Database Table: CENSUS_DATA) <br>
This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.
https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

   A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:   https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2
   

2. Chicago Public Schools Subset (Database Table: CHICAGO_PUBLIC_SCHOOLS) <br>
This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. 
https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

   A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at: https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t
   

3. Chicago Crime Data Subset (Database Table: CHICAGO_CRIME_DATA) <br>
This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.
https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

   A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at: https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2


# Connect to the database


In [1]:
# These libraries are pre-installed in SN Labs (Skills Networks Labs). 
# Since this project was completed in the SN labs, the following codes were commented out and preserved here for future use. 

# If running in another environment please uncomment lines below to install them:
# !pip install --force-reinstall ibm_db==3.1.0 ibm_db_sa==0.3.3
# Ensure we don't load_ext with sqlalchemy>=1.4 (incompadible)
# !pip uninstall sqlalchemy==1.4 -y && pip install sqlalchemy==1.3.24
# !pip install ipython-sql

In [2]:
# Load the SQL extension 
%load_ext sql

In [3]:
# Establish a connection with the database
# Note: The connection string is of the format:
# %sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL
# Enter the connection string for my Db2 on Cloud database instance below to connect to the database.

%sql ibm_db_sa://yfw87399:57NZcUc4XIy6Mtah@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB?security=SSL

'Connected: yfw87399@BLUDB'


# SQL Queries


In [20]:
# Find the total number of crimes recorded in the CRIME table.

%%sql
SELECT COUNT(*) AS Total_NUMBER_OF_CRIMES 
FROM CHICAGO_CRIME_DATA;

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[(533,)]

In [16]:
# List community areas with per capita income less than 11000.

%%sql
SELECT COMMUNITY_AREA_NAME, PER_CAPITA_INCOME 
FROM CENSUS_DATA 
WHERE PER_CAPITA_INCOME < 11000;

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('West Garfield Park', 10934),
 ('South Lawndale', 10402),
 ('Fuller Park', 10432),
 ('Riverdale', 8201)]

In [7]:
# List all case numbers for crimes involving minors?(Children are not considered minors for the purposes of crime analysis.)

%%sql 
SELECT CASE_NUMBER
FROM CHICAGO_CRIME_DATA 
WHERE DESCRIPTION LIKE '%MINOR%';

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('HL266884',), ('HK238408',)]

In [8]:
# List kidnapping crimes involving a child.

%%sql 
SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION
FROM CHICAGO_CRIME_DATA
WHERE PRIMARY_TYPE = 'KIDNAPPING' AND DESCRIPTION LIKE '%CHILD%';

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('HN144152', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

In [9]:
# The types of crimes that were recorded at schools.

%%sql
SELECT DISTINCT(PRIMARY_TYPE)
FROM CHICAGO_CRIME_DATA
WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%';

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('ASSAULT',),
 ('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('CRIMINAL TRESPA',),
 ('NARCOTICS',),
 ('PUBLIC PEACE VI',)]

In [10]:
# List the average safety score for all types of schools.

%%sql
SELECT "Elementary, Middle, or High School", AVG(SAFETY_SCORE) AS AVERAGE_SAFETY_SCORE
FROM CHICAGO_PUBLIC_SCHOOLS
GROUP BY "Elementary, Middle, or High School";

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('ES', 49), ('HS', 49), ('MS', 48)]

In [11]:
# List 5 community areas with highest % of households below poverty line.

%%sql
SELECT COMMUNITY_AREA_NAME, PERCENT_HOUSEHOLDS_BELOW_POVERTY
FROM CENSUS_DATA
ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC
LIMIT 5 ;

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('Riverdale', Decimal('56.5')),
 ('Fuller Park', Decimal('51.2')),
 ('Englewood', Decimal('46.6')),
 ('North Lawndale', Decimal('43.1')),
 ('East Garfield Park', Decimal('42.4'))]

In [21]:
# Identify community area number of the community area that is most crime prone. 
%%sql
SELECT COMMUNITY_AREA_NUMBER, COUNT(COMMUNITY_AREA_NUMBER) AS NUMBER_OF_CRIMES
FROM CHICAGO_CRIME_DATA
GROUP BY COMMUNITY_AREA_NUMBER
ORDER BY COUNT(COMMUNITY_AREA_NUMBER) DESC
LIMIT 1;

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[(25, 43)]

In [13]:
# Use a sub-query to find the name of the community area with highest hardship index.

%%sql
SELECT COMMUNITY_AREA_NAME 
FROM CENSUS_DATA
WHERE HARDSHIP_INDEX = (SELECT MAX(HARDSHIP_INDEX) FROM CENSUS_DATA);

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('Riverdale',)]

In [19]:
# Use a sub-query to find the Community Area Name with the highest number of crimes.

%%sql
SELECT COMMUNITY_AREA_NAME
FROM CENSUS_DATA
WHERE COMMUNITY_AREA_NUMBER = (
    SELECT COMMUNITY_AREA_NUMBER
    FROM CHICAGO_CRIME_DATA AS CCD
    GROUP BY CCD.COMMUNITY_AREA_NUMBER
    ORDER BY COUNT(CCD.COMMUNITY_AREA_NUMBER) DESC LIMIT 1);

 * ibm_db_sa://yfw87399:***@b1bc1829-6f45-4cd4-bef4-10cf081900bf.c1ogj3sd0tgtu0lqde00.databases.appdomain.cloud:32304/BLUDB
Done.


[('Austin',)]


# Findings and Implications


1. The total number of crimes occurred during the period covered by the dataset is 533 cases. 
2. Four community areas out of seventy-seven community areas (per the Chicago Census Data dataset) have per capita income less than 11000 dollars. Thus, the city of Chicago as a whole is not an extremely poor area. 
3. There are two cases of crimes involving minors. Therefore, the majority of the 533 crimes were committed by adults. 
4. One kidnapping crime involved a child. This means that kidnapping a child is rare in the city of Chicago. 
5. Six types of crimes that were recorded at schools. However, none is considered as a serious crime type per classifications of crimes. 
6. The average safety score of all types of schools is about 48-49 on a scale of 0-100. This means that on average, the security measures implemented at schools are not as sufficient as they should be. 
7. In the five community areas that have highest percentage of households below poverty line, about 42.4%-56.5% of the households is in poverty. Therefore, although generally, Chicago is not regarded as a poor city, there still are quite a few households in certain areas of Chicago that are poverty-stricken. 
8. The community with the community area of 25 is most crime prone. It has 43 crimes recorded. A later query identified that the name of this community is Austin (see 10.). Hence, for those people who are planning to reside near or in Austin might want to consider this finding and act accordingly. 
9. Riverside, which is a very poor community, with a 56.5% of households below poverty line (see query result of Finding 7. above) and an income per capita of 8201 (see query result of Finding 2 above), has the highest hardship index. Therefore, Riverside definitely is the poorest community of all.
10. Based on the query of finding the Community Area Name with the highest number of crimes, Austin was found to have the highest number of crimes. This identification was completed by first finding the Community Area Number (25) in the Chicago_Crime_Data Table), and then using it to search for the Community Area Name in the Chicago_Census_Data Table, which contains both Community Area Name and Community Area Number. This process differs from the query of obtaining Finding 8, in which the query was performed using solely the Chicago_Crime_Data Table, which contains only the Community Area Number. Therefore, the Community Area Number was available in Finding 8, but not the Community Area Name. 
